DataLoader.lua

```diff
@@ -18,11 +28,17 @@ function DataLoader:sub(start, stop, inputs, targets, ...)
 end
 
 function DataLoader:shuffle()
-   error"Not Implemented"
+   self._permuteIndex = torch.randperm(self.imagePath:size(1)):long()
 end
 
 function DataLoader:split(ratio)
```

Comment:
This implementation won't work for all sub-classes. `imagePath` is only supported by `ImageClass`.